### PR TITLE
docs: add documentation for ambassador traffic router

### PR DESCRIPTION
**Pull request by leoluz**

### CODE REVIEW

File: docs/features/traffic-management/ambassador.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
+# Ambassador Edge Stack
+
+Ambassador Edge Stack provides the functionality you need at the edge your Kubernetes cluster (hence, an "edge stack"). This includes an API gateway, ingress controller, load balancer, developer portal, canary traffic routing and more. It provides a group of CRDs that users can configure to enable different functionalities.
```
Comment (richarddli) [inline docs/features/traffic-management/ambassador.md]: We should add a link to https://www.getambassador.io/products/edge-stack/ above

File: docs/features/traffic-management/ambassador.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
+# Ambassador Edge Stack
+
+Ambassador Edge Stack provides the functionality you need at the edge your Kubernetes cluster (hence, an "edge stack"). This includes an API gateway, ingress controller, load balancer, developer portal, canary traffic routing and more. It provides a group of CRDs that users can configure to enable different functionalities. 
+
+Argo-Rollouts provides an integration that leverages Ambassador's [canary routing capability](https://www.getambassador.io/docs/pre-release/topics/using/canary/). This allows the traffic to your application to be gradually incremented while new versions are being deployed.
```
Comment (richarddli) [inline docs/features/traffic-management/ambassador.md]: Let's link to latest instead of pre-release

File: docs/features/traffic-management/ambassador.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+spec:
+  prefix: /someapp
+  rewrite: /
+  service: someapp-canary:80
+  weight: 30
+```
+
+In the example above we are configuring Ambassador to route 30% of the traffic coming from `<public ingress>/someapp` to the service `someapp-canary` and the rest of the traffic will go to the service `someapp-stable`. If users want to gradually increase the traffic to the canary service, they have to update the `canary-mapping` setting the weight to the desired value either manually or automating it somehow. 
+
+With Argo-Rollouts there is no need to create the `canary-mapping`. The process of creating it and gradually update its weight is fully automated by the Argo-Rollouts controller. The following example shows how to configure the `Rollout` resource to use Ambassador as a traffic router for canary deployments:
```
Comment (richarddli) [inline docs/features/traffic-management/ambassador.md]: s/update/updating

File: docs/features/traffic-management/ambassador.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+      - setWeight: 30
+      - pause: {duration: 60s}
+      - setWeight: 60
+      - pause: {duration: 60s}
+```
+
+Under `spec.strategy.canary.trafficRouting.ambassador` there are 2 possible attributes:
+
+- `apiVersion`: Optional. If you are using an older version of Ambassador you can specify its `apiVersion` (e.g.: `getambassador.io/v1`). If not provided, Argo-Rollouts will use the default value as `getambassador.io/v2`
+- `mappings`: Required. If your application exposes 2 different ports for different servers (e.g.: REST and gRPC) you can provide the stable mappings in this list and Argo-Rollouts will create the canary mappings for both of them. At least one mapping is necessary to be provided. If no mapping is provided Argo-Rollouts will send an error event and the rollout will be aborted.
```
Comment (richarddli) [inline docs/features/traffic-management/ambassador.md]: I'd reword:

At least one Ambassador mapping that Argo Rollouts will use to manage the canary rollouts. Multiple mappings are also supported in case there are multiple routes to the service (e.g., your service has multiple ports, or can be accessed via different URLs).

File: docs/features/traffic-management/ambassador.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+When Ambassador is configured in the `trafficRouting` attribute of the manifest, the Rollout controller will:
+1. Create one canary mapping for each stable mapping provided in the Rollout manifest
+1. Proceed with the steps according to the configuration updating the canary mapping weight
+1. At the end of the process Argo-Rollout will delete all the canary mappings created
+
+## Endpoint Resolver
+
+Argo-Rollout will dynamically modify the `Service` selectors when the rollout starts and when it concludes in order to do version promotion. Ambassador mappings use service resolver by default to route traffic to Pods. However when the connection is opened, changes in the service selectors aren't directly reflected and traffic can be sent to old pods if they are still alive. For this reason, it is important to configure Ambassador mappings to use endpoint resolver to avoid this problem.
```
Comment (richarddli) [inline docs/features/traffic-management/ambassador.md]: I would reword because it's unlikely that Argo users know the difference between service/endpoint resolvers.

"By default, Ambassador uses `kube-proxy` to route traffic to Pods. However, this does provide true L7 load balancing which is desirable in a canary workflow. Thus, we configure Ambassador to bypass `kube-proxy` and route traffic directly to pods. This approach is called [endpoint routing](https://www.getambassador.io/docs/latest/topics/running/load-balancer/).

File: docs/features/traffic-management/ambassador.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+apiVersion: getambassador.io/v2
+kind:  Mapping
+metadata:
+  name: stable-mapping
+spec:
+  resolver: endpoint
+  prefix: /someapp
+  rewrite: /
+  service: someapp-stable:80
+```
```
Comment (richarddli) [inline docs/features/traffic-management/ambassador.md:93]: Add at the end:

"For more details about the Ambassador and Argo Rollouts integration, see the [Ambassador Argo documentation](https://deploy-preview-508--datawire-ambassador.netlify.app/docs/pre-release/argo/).

File: docs/getting-started/ambassador/index.md [hunk 7]
```diff
@@ -0,0 +1,236 @@
+# Argo Rollouts and Ambassador Quick Start
+
+This tutorial will walk you through the process of configuring Argo Rollouts to work with Ambassador to facilitate canary releases. All files used in this guide are available in the [examples](https://github.com/argoproj/argo-rollouts/blob/master/examples/ambassador) directory of this repository.
+
+## Requirements
+
+- Kubernetes cluster
+- Argo rollouts installed in the cluster
```
Comment (richarddli) [inline docs/getting-started/ambassador/index.md]: Rollouts

File: docs/getting-started/ambassador/index.md [hunk 8]
```diff
@@ -0,0 +1,236 @@
...
+```
+
+In your other terminal window, you can verify that the canary is progressing appropriately by sending requests in a loop:
+
+```
+while true; do curl -k https://$AMBASSADOR_LB_ENDPOINT/echo/; sleep 0.2; done
+```
+
+This will display a running list of responses from the service that will gradually transition from VERSION 1 strings to VERSION 2 strings.
+
```
Comment (richarddli) [inline docs/getting-started/ambassador/index.md:236]: Add:

"For more details about the Ambassador and Argo Rollouts integration, see the Ambassador Argo documentation. (https://deploy-preview-508--datawire-ambassador.netlify.app/docs/pre-release/argo/)

**Review by richarddli — CHANGES_REQUESTED** [review]

File: docs/features/traffic-management/ambassador.md (hunk 1 above)
Comment (leoluz) [reply docs/features/traffic-management/ambassador.md]: done

File: docs/features/traffic-management/ambassador.md (hunk 2 above)
Comment (leoluz) [reply docs/features/traffic-management/ambassador.md]: done

File: docs/features/traffic-management/ambassador.md (hunk 3 above)
Comment (leoluz) [reply docs/features/traffic-management/ambassador.md]: done

File: docs/features/traffic-management/ambassador.md (hunk 4 above)
Comment (leoluz) [reply docs/features/traffic-management/ambassador.md]: done

File: docs/getting-started/ambassador/index.md (hunk 7 above)
Comment (leoluz) [reply docs/getting-started/ambassador/index.md]: done

File: docs/getting-started/ambassador/index.md (hunk 8 above)
Comment (leoluz) [reply docs/getting-started/ambassador/index.md:236]: Is this the final link?

Comment (richarddli) [reply docs/getting-started/ambassador/index.md:236]: use the same link as above comment that has identical text

File: docs/features/traffic-management/ambassador.md (hunk 6 above)
Comment (leoluz) [reply docs/features/traffic-management/ambassador.md:93]: done

File: docs/features/traffic-management/ambassador.md (hunk 5 above)
Comment (leoluz) [reply docs/features/traffic-management/ambassador.md]: This is better. Thank you. I had to adapt a few things in the text to fit in the context of the document. Let me know if you think it should be modified.

File: docs/getting-started/ambassador/index.md (hunk 8 above)
Comment (leoluz) [reply docs/getting-started/ambassador/index.md:236]: ok. done

**Review by richarddli — APPROVED** [review]
